mlfinlab/labeling/matrix_flags.py

```diff
@@ -0,0 +1,110 @@
+        percentile_cutoffs = []
+        for i in [10, 20, 30, 40, 50, 60, 70, 80, 90, 100]:
+            percentile_cutoffs.append(np.percentile(data_window, i))
+
+        # Each value in bins corresponds to the value of the same index in data_window. If bin is 0, the corresponding
+        # point in data_window is in the 0-10th percentile, if 1 then 10-20th percentile, ..., if 9 then 90-100th
+        # percentile.
+        bins = np.digitize(data_window, bins=percentile_cutoffs, right=True)
+
+        # Dictionary to map values in data window to their percentile bin.
+        value_to_bin = dict(zip(data_window, bins))
+
+        # Split data window into 10 chronological sub-windows.
+        data_split = np.array_split(data_window, 10)
+
+        # Create the matrix, one column at a time from left to right. The leftmost column corresponds to the first
+        # tenth of data, and the rightmost corresponds to the final tenths. The top element of each column corresponds
+        # to the count in the tenth in the 90-100th percentile of data window, second from top is count in the tenth
+        # in the 80-90th percentile and so on until the last element is the count in the 0-10th percentile.
+        matrix = pd.DataFrame()
+        col_num = 0  # Left most column (out of 10) to populate using the first tenth of data.
+        for tenth in data_split:
+            # Apply the dictionary to data to get the bins each data point belongs to.
+            tenth_bins = np.vectorize(value_to_bin.get)(tenth)
+            # We count the number of times each bin appears in each tenth of data, starting from the highest decile.
+            column = [np.count_nonzero(tenth_bins == 9), np.count_nonzero(tenth_bins == 8),
+                      np.count_nonzero(tenth_bins == 7), np.count_nonzero(tenth_bins == 6),
+                      np.count_nonzero(tenth_bins == 5), np.count_nonzero(tenth_bins == 4),
+                      np.count_nonzero(tenth_bins == 3), np.count_nonzero(tenth_bins == 2),
+                      np.count_nonzero(tenth_bins == 1), np.count_nonzero(tenth_bins == 0)]
+            matrix[col_num] = column
+            col_num += 1
+
+        return matrix
+
+    def apply_template_to_matrix(self, matrix, template):
+        """
+        :param matrix: (pd.DataFrame) Processed 10 by 10 matrix, where each column represents a chronological tenth
+                        of the data, and each row represents a decile relative to the entire data window.
+        :param template: (pd.DataFrame) Template to apply the processed matrix to.
+        :return: (float) The total score for the day. Consists of the sum of sum of columns of the result from
+                    multiplying the matrix element-wise with the template.
+        """
+        new_mat = matrix * template
+        total_fit = sum(new_mat.sum(axis=0))
+        return total_fit
+
+    def apply_labeling_matrix(self):
```

Comment:
Which of these methods should not be called by the end user? They should be hiden by adding a single underscore to the name. Example: `_foo()`